packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -183,6 +196,7 @@ realm_on_object_change_func_t get_on_object_change() {
         jenv->CallVoidMethod(static_cast<jobject>(userdata),
                              on_change_method,
                              reinterpret_cast<jlong>(changes));
+        jni_check_exception(jenv);
```

Comment:
I don't think we have any tests for these, but then again, I suspect it will also be very tricky do to since it is happening inside the Notifier?

Reply:
Yes, it would be tricky to test that we catch this, to do so, there should be an issue at [object change callback](https://github.com/realm/realm-kotlin/blob/be75e66560737e6b4b13be4941f9ac3f09e02747/packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/SuspendableNotifier.kt#L121) with the realm/changeflow.